pages/ekmann.js

```diff
@@ -0,0 +1,352 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { Container, Page } from "@/components/Layout/Layout.styled";
+import NavButton from "@/components/NavButton/NavButton";
+import Circle from "@/components/Circle/Circle";
+import { TbList } from "react-icons/tb";
+
+function EmotionAnalysis() {
+  const [formData, setFormData] = useState({
+    0: "",
+    1: "",
+    2: "",
+    3: "",
+    4: "",
+    5: "",
+    6: "",
+    7: "",
+    8: "",
+    9: "",
+  });
+  const [emotionResult, setEmotionResult] = useState("");
+  const [language, setLanguage] = useState("en");
+  const [page, setPage] = useState(0);
+  const [predictionsState, setPredictionsState] = useState(false);
+  const [showList, setShowList] = useState(false);
+  const [emptyFieldsError, setEmptyFieldsError] = useState("");
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const handleLanguageSelect = (event) => {
+    if (event.target.value == "english") {
+      setLanguage("en");
+    }
+    if (event.target.value == "german") {
+      setLanguage("de");
+    }
+    if (event.target.value == "espanol") {
+      setLanguage("es");
+    }
+  };
+
+  const handleSubmit = async (event) => {
+    // Check each field in formData for emptiness
+    for (const key in formData) {
+      if (formData.hasOwnProperty(key)) {
+        const value = formData[key].trim(); // Trim whitespace from value
+        if (!value) {
+          // If any field is empty, log error and exit early
+          setEmptyFieldsError(
+            `Field '${key}' is empty. Please enter some text in "EVERY" field.`
+          );
+
+          return;
+        }
+      }
+    }
+    event.preventDefault();
+    //here you can log the data which is saved before it i processed very well
+    const inputText = JSON.stringify(formData);
+
+    const url =
+      "https://ekman-emotion-analysis.p.rapidapi.com/ekman-emotion?all=true";
+    //?all=true for all emotions without filter
+    const options = {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        Accept: "application/json",
+        "X-RapidAPI-Key": "d494478f31mshab8f5690c7d60c8p1f2035jsna0a331ad02cd",
+        "X-RapidAPI-Host": "ekman-emotion-analysis.p.rapidapi.com",
```

Comment:
it is really important to move the API key out of this file and into a secure file like the .env.local file. Otherwise this is a big security issue!